validation/datehelpers.go

```diff
@@ -0,0 +1,80 @@
+package validation
+
+import (
+	"regexp"
+	"strconv"
+	"time"
+
+	errorhandler "github.com/cloudtrust/common-service/errors"
+)
+
+type longDuration struct {
+	years  int
+	months int
+	days   int
+}
+
+const (
+	// RegExpLongDuration is a regular expression used to parse long durations
+	//RegExpLongDuration = `^((([1-9]\d*)([ymwd]))+)$`
+	RegExpLongDuration = `([1-9]\d*)([ymwd])`
+)
+
+var (
+	regExpLongDuration = regexp.MustCompile(RegExpLongDuration)
+)
+
+func parseLongDuration(value string) (longDuration, error) {
+	var values = regExpLongDuration.FindAllStringSubmatch(value, -1)
+	var duration longDuration
+	var totalLen = len(value)
+	if totalLen == 0 {
+		return longDuration{}, errorhandler.CreateMissingParameterError("duration")
+	}
+	for _, extract := range values {
+		totalLen -= len(extract[0])
+		var number, _ = strconv.Atoi(extract[1])
+		switch extract[2] {
+		case "y":
+			duration.years += number
+			break
+		case "m":
+			duration.months += number
+			break
+		case "w":
+			duration.days += 7 * number
+			break
+		case "d":
+			duration.days += number
+			break
+		}
+	}
+	if totalLen != 0 {
```

Comment:
I would suggest to first validate if it is a valid duration string, and then process it. 
It really took me a while to understand this mechnaism with totalLen, but maybe it is just me.

Reply:
In most cases, this value should be valid so it will be faster to operate like this.
Ideally, I would prefer to have a single RegExp which would validate and extract all values but I did not achieve to write it this way.